sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala

```diff
@@ -50,7 +50,7 @@ object ExpressionSet {
 class ExpressionSet protected(
     protected val baseSet: mutable.Set[Expression] = new mutable.HashSet,
     protected val originals: mutable.Buffer[Expression] = new ArrayBuffer)
-  extends Set[Expression] {
+  extends Set[Expression] with Serializable {
```

Comment:
Thank you for your explanation on how ```constraints``` is accessed. From the implementation view, my understanding is that it is still necessary  to declare ```Serializable``` for ```ExpressionSet``` . Is there another good idea to enable ```Serializable``` only for ```LogicalPlan```.